source/develop/release-management/features/virt/engine-dedicated-cpus.md

```diff
@@ -0,0 +1,102 @@
+---
+Title: Engine Dedicated CPUs
+Category: feature
+Authors: sabusale
+---
+
+# Engine Dedicated CPUs
+
+## Summary
+
+Once choosing dedicated CPU policy, The guest vCPUs will be strictly pinned to a set of host pCPUs
+(similarly to static CPU pinning). The set of pCPUs will be chosen to match the required guest CPU topology.
+
+Engine handles the pinning automatically when starting the VM as opposed to the manual pinning that is
+done by the user and is static.
+
+## Owner
+
+-   Name: Saif Abu Saleh (sabusale)
+-   Email: sabusale@redhat.com
+
+## Responsibilities
+
+VDSM:
+
+- Manages the available pinned CPUs
+
+Engine:
+- Schedule VM to run on best satisifing host
+- creates vCpu to pCpu mappings
+
+
+## Backend
+
+1. Add dedicated to CPU Pinning Policy enum
+
+2.  Scheduling
+    1. Start VM:
+        - Scheduling phase
+            - Score all hosts to find best matching host with enough pCPUs
+        - Run phase
+            - Create vCpu to pCpu mapping based on the host free pCPUs reported
```

Comment:
`based on the host free pCPUs reported` - not sure what you mean by `free pCPUs`, whether it means that it is not exclusively pinned or that no vCPU runs on it - the latter is not useful at all since it may change right after the report
so if we're only interested in the former, we don't have to rely on what the host report - that's information the engine produces (besides the "reserved cpus" that need to be taken into account)